app/classes/query/modules/ordering.rb

```diff
@@ -1,121 +1,219 @@
 # frozen_string_literal: true
 
-module Query::Modules::Ordering
-  def initialize_order
-    by = params[:by]
-    # Let queries define custom order spec in "order", but have explicitly
-    # passed-in "by" parameter take precedence.  If neither is given, then
-    # fall back on the "default_order" finally.
-    return unless by || order.blank?
-
-    by ||= default_order
-    by = by.dup
-    reverse = !!by.sub!(/^reverse_/, "")
-    result = initialize_order_specs(by)
-    self.order = reverse ? reverse_order(result) : result
-  end
+module Query
+  module Modules
+    module Ordering
+      def initialize_order
+        by = params[:by]
+        # Let queries define custom order spec in "order", but have explicitly
+        # passed-in "by" parameter take precedence.  If neither is given, then
+        # fall back on the "default_order" finally.
+        return unless by || order.blank?
+
+        by ||= default_order
+        by = by.dup
+        reverse = !!by.sub!(/^reverse_/, "")
+        result = initialize_order_specs(by)
+        self.order = reverse ? reverse_order(result) : result
+      end
+
+      def initialize_order_specs(by)
+        sorting_method = "sort_by_#{by}"
+        unless ::Query::Modules::Ordering.private_method_defined?(
+          sorting_method
+        )
+          raise(
+            "Can't figure out how to sort #{model.name.pluralize} by :#{by}."
+          )
+        end
+
+        send(sorting_method, model)
+      end
+
+      #######
+
+      private
+
+      def sort_by_updated_at(model)
+        return unless model.column_names.include?("updated_at")
+
+        "#{model.table_name}.updated_at DESC"
+      end
+
+      def sort_by_created_at(model)
+        return unless model.column_names.include?("created_at")
+
+        "#{model.table_name}.created_at DESC"
+      end
+
+      def sort_by_last_login(model)
+        return unless model.column_names.include?("last_login")
 
-  def initialize_order_specs(by)
-    table = model.table_name
-    columns = model.column_names
-    case by
+        "#{model.table_name}.last_login DESC"
+      end
 
-    when "updated_at", "created_at", "last_login", "num_views"
-      "#{table}.#{by} DESC" if columns.include?(by)
+      def sort_by_num_views(model)
+        return unless model.column_names.include?("num_views")
+
+        "#{model.table_name}.num_views DESC"
+      end
+
+      def sort_by_date(model)
+        if model.column_names.include?("date")
+          "#{model.table_name}.date DESC"
```

Comment:
@pellaea, @nimmolo:
I think the first condition should be deleted because there is no table with a `date` column.
Am I missing something?

Reply:
Sounds right.
It seems like it would make sense then to also change the name of the sorting condition to `by: :when`, but it's not necessary.

Reply:
That would work, but I think would necessitate more changes elsewhere, which I'd rather avoid in this PR.

Reply:
Yes, let's please resist making unnecessary changes to the public interface! :)